lib/templates.php

```diff
@@ -232,3 +225,95 @@ function filter_rest_wp_template_query( $args, $request ) {
 	return $args;
 }
 add_filter( 'rest_wp_template_query', 'filter_rest_wp_template_query', 99, 2 );
+
+/**
+ * Creates a template auto-draft if it doesn't exist yet.
+ *
+ * @access private
+ *
+ * @param string $slug    Template slug.
+ * @param string $theme   Template theme.
+ * @param string $content Template content.
+ */
+function create_auto_draft_for_template( $slug, $theme, $content ) {
+	// We check if an auto-draft was already created,
+	// before running the REST API calls
+	// because the site editor needs an existing auto-draft
+	// for each theme template to work properly.
+	$template_query = new WP_Query(
+		array(
+			'post_type'      => 'wp_template',
+			'post_status'    => array( 'publish', 'auto-draft' ),
+			'title'          => $slug,
+			'meta_key'       => 'theme',
+			'meta_value'     => $theme,
+			'posts_per_page' => 1,
+			'no_found_rows'  => true,
+		)
+	);
+	$template_post  = $template_query->have_posts() ? $template_query->next_post() : null;
+	if ( ! $template_post ) {
+		wp_insert_post(
+			array(
+				'post_content' => $content,
+				'post_title'   => $slug,
+				'post_status'  => 'auto-draft',
+				'post_type'    => 'wp_template',
+				'post_name'    => $slug,
+				'meta_input'   => array(
+					'theme' => $theme,
```

Comment:
This post-meta is not used or needed for templates. I assume this was copied and pasted from the template part code and adding it here was unintentional?

Reply:
It is intended and I believe it's important to know from which theme a template is coming in order to avoid weird issues (mixing templates) when switching themes.

Reply:
This also simplifies the theme exports a lot.

Reply:
It wasn't needed before, as far as I'm aware. If it's needed for this implementation, that only tells me that this is a cumbersome implementation, because it comes at the cost of cluttering the database with content from deactivated themes. #26383 is a much more lightweight implementation that successfully handles switching themes without using a theme post-meta.